include/motor-control/core/motor_interrupt_handler.hpp

```diff
@@ -37,45 +37,85 @@ namespace motor_handler {
  * GenericQueue -> A FreeRTOS queue of any shape.
  */
 
+constexpr const int clk_frequency = 85000000 / (5001 * 2);
+
 template <template <class> class QueueImpl>
 requires MessageQueue<QueueImpl<Move>, Move>
 class MotorInterruptHandler {
   public:
     using GenericQueue = QueueImpl<Move>;
+    bool has_active_move = false;
 
     MotorInterruptHandler() {}
+    MotorInterruptHandler& operator=(MotorInterruptHandler&) = delete;
+    MotorInterruptHandler&& operator=(MotorInterruptHandler&&) = delete;
+    MotorInterruptHandler(MotorInterruptHandler&) = delete;
+    MotorInterruptHandler(MotorInterruptHandler&&) = delete;
 
     void set_message_queue(GenericQueue* g_queue) { queue = g_queue; }
 
     bool has_messages() { return queue->has_message_isr(); }
 
-    bool can_step() { return (has_active_move && inc < buffered_move.steps); }
+    bool can_step() { return (step_count < buffered_move.target_position); }
+
+    bool tick() {
+        sq32_31 old_step_count = step_count;
+        step_count += static_cast<sq32_31>(steps_per_tick);
+        return bool((old_step_count ^ step_count) & tick_flag);
+    }
 
     void update_move() {
-        finish_current_move();
-        get_move(&buffered_move);
+        has_active_move = queue->try_read_isr(&buffered_move);
+        // (TODO: lc) We should check the direction (and set respectively)
+        // the direction pin for the motor once a move is being pulled off the
+        // queue stack. We'll probably want to think about moving the hardware
+        // pin configurations out of motion controller.
     }
 
     void finish_current_move() {
+        // (TODO: lc) We need to handle overflow cases on the position
+        // tracker.
+        position_tracker += step_count;
         has_active_move = false;
         buffered_move = Move{};
-        inc = 0x0;
+        step_count = 0x0;
     }
 
-    void increment_counter() { inc++; }
+    bool pulse() {
```

Comment:
What does this function do? It's very complex and it's name is not descriptive. 

What does the return value mean?